a3fe/configuration/engine_config.py

```diff
@@ -157,141 +136,123 @@ def _check_cutoff_values(self):
+            Custom lambda array (if not using default values)
+        lambda_val : Optional[float]
+            Current lambda value
         """
-        config_filename = self.get_file_name()
-        # First, generate the configuration string
         config_lines = [
             "### Integrator ###",
-            f"nmoves = {self.nmoves}",
             f"timestep = {self.timestep} * femtosecond",
+            #f"runtime = {self.runtime} * nanosecond",
+            f"nmoves = {self.nmoves}",
             f"constraint = {self.constraint}",
             f"hydrogen mass repartitioning factor = {self.hydrogen_mass_factor}",
             f"integrator = {self.integrator}",
             f"inverse friction = {self.inverse_friction} * picosecond",
             f"temperature = {self.temperature} * celsius",
             f"thermostat = {self.thermostat}",
-            "",
-            "### Barostat ###",
+            "\n### Barostat ###",
             f"barostat = {self.barostat}",
             f"pressure = {self.pressure} * atm",
-            "",
-            "### Non-Bonded Interactions ###",
+            "\n### Non-Bonded Interactions ###",
             f"cutoff type = {self.cutoff_type}",
             f"cutoff distance = {self.cutoff_distance} * angstrom",
         ]
         if self.cutoff_type == "cutoffperiodic" and self.reaction_field_dielectric is not None:
             config_lines.append(f"reaction field dielectric = {self.reaction_field_dielectric}")    
 
         config_lines.extend([
-            "",
-            "### Trajectory ###",
+            "\n### Trajectory ###",
             f"buffered coordinates frequency = {self.buffered_coords_freq}",
             f"center solute = {self.center_solute}",
-            "",
-            "### Minimisation ###",
+            "\n### Minimisation ###",
             f"minimise = {self.minimise}",
-            "",
-            "### Alchemistry ###",
+            "\n### Alchemistry ###",
             f"perturbed residue number = {self.perturbed_residue_number}",
             f"energy frequency = {self.energy_frequency}",
             f"ligand charge = {self.ligand_charge}",
-            f"charge difference = {self.charge_difference}",
-            "",
-            "### Restraints ###",
+            "\n### Restraints ###",
             f"use boresch restraints = {self.use_boresch_restraints}",
             f"turn on receptor-ligand restraints mode = {self.turn_on_receptor_ligand_restraints}"
         ])
         # 2) Lambda parameters
-        config_lines.extend(["", "### Lambda / Alchemical Settings ###"])
-        
-        if self.lambda_array:
-            lambda_str = f"[{', '.join(str(x) for x in self.lambda_array)}]"
-            config_lines.append(f"lambda_array = {lambda_str}")
-        if self.lambda_val is not None:
-            config_lines.append(f"lambda_val = {self.lambda_val}")
+        config_lines.extend(["", "### Lambda ###"])
+
+        # Determine lambda array
```

Comment:
Validation of lambda array should happen with a validator - no need for this here. Thanks